pkg/archive/archive_unix.go

```diff
@@ -96,7 +91,12 @@ func handleTarTypeBlockCharFifo(hdr *tar.Header, path string) error {
 		mode |= unix.S_IFIFO
 	}
 
-	return system.Mknod(path, mode, int(system.Mkdev(hdr.Devmajor, hdr.Devminor)))
+	err := system.Mknod(path, mode, int(system.Mkdev(hdr.Devmajor, hdr.Devminor)))
+	if sys.RunningInUserNS() {
```

Comment:
is there a specific error returned in here?

Reply:
Mknodding device nodes inside UserNS fails in most cases

Reply:
I think you usually get `EPERM`.

Reply:
updated